### PR TITLE
[WWST-889] Danalock V3 Z-Wave genericized DTH with no codes support.

### PR DESCRIPTION
**Pull request by ALamchaS**
This can work with generic zwave-lock DTH with some modification. It is significantly smaller - close to 1/3rd of generic zwave-lock DTH size.

Base DTH is huge, so I may have missed some useless code. Please feel free to comment on everything, @greens , @dkirker , @mckeed 

cc for SRPOL PE:
@MMateusiakS @DCzarneckaS  @MAblewiczS  @KKlimczukS

### CODE REVIEW

**Comment by greens** [comment]
I think the problem with putting it in the generic lock DTH is that it would get picked up automatically by lock code manager.

File: devicetypes/smartthings/zwave-lock-without-codes.src/zwave-lock-without-codes.groovy [hunk 1]
```diff
@@ -0,0 +1,593 @@
...
+	def hubAction = null
+	try {
+		def cmds = []
+		if (!device.currentState("lock") || !state.configured) {
+			log.debug "Returning commands for lock operation get and battery get"
+			if (!state.configured) {
+				cmds << doConfigure()
+			}
+			cmds << refresh()
+			if (!state.MSR) {
```
Comment (greens) [inline devicetypes/smartthings/zwave-lock-without-codes.src/zwave-lock-without-codes.groovy]: I think this is a holdover from an older way of doing things. We can probably just use `zwaveInfo.mfr` for anything we would be using this for if we're starting from scratch.

File: devicetypes/smartthings/zwave-lock-without-codes.src/zwave-lock-without-codes.groovy [hunk 2]
```diff
@@ -0,0 +1,593 @@
...
+				cmds << zwave.manufacturerSpecificV1.manufacturerSpecificGet().format()
+			}
+			if (!state.fw) {
```
Comment (greens) [inline devicetypes/smartthings/zwave-lock-without-codes.src/zwave-lock-without-codes.groovy]: nothing is ever done with this so we can probably leave it out

File: devicetypes/smartthings/zwave-lock-without-codes.src/zwave-lock-without-codes.groovy [hunk 3]
```diff
@@ -0,0 +1,593 @@
...
+
+/**
+ * Returns the list of commands to be executed when the device is being configured/paired
+ */
+def doConfigure() {
+	log.trace "[DTH] Executing 'doConfigure()' for device ${device.displayName}"
+	state.configured = true
+	def cmds = []
+	cmds << secure(zwave.doorLockV1.doorLockOperationGet())
+	if (zwaveInfo.mfr != "010E") {
```
Comment (greens) [inline devicetypes/smartthings/zwave-lock-without-codes.src/zwave-lock-without-codes.groovy]: Why restrict this based on mfr?

File: devicetypes/smartthings/zwave-lock-without-codes.src/zwave-lock-without-codes.groovy (hunk 1 above)
Comment (ALamchaS) [reply devicetypes/smartthings/zwave-lock-without-codes.src/zwave-lock-without-codes.groovy]: Ok. Thanks. I removed this.

File: devicetypes/smartthings/zwave-lock-without-codes.src/zwave-lock-without-codes.groovy (hunk 2 above)
Comment (ALamchaS) [reply devicetypes/smartthings/zwave-lock-without-codes.src/zwave-lock-without-codes.groovy]: Removed. Thanks.

**Review by greens — APPROVED** [review]

**Pull request merged**